pandas/core/groupby.py

```diff
@@ -1923,8 +1923,16 @@ def __init__(self, index, grouper=None, obj=None, name=None, level=None,
 
             # a passed Categorical
             elif isinstance(self.grouper, Categorical):
+
+                # must have an ordered categorical
+                if self.sort:
+                    if not self.grouper.ordered:
+                        raise ValueError("cannot sort by an unordered Categorical in the grouper\n"
```

Comment:
I haven't read the rest of the file but IMO these two should not raise:

* using a string column as the group_by -> If the column is converted to a Categorical first and the default is kept, then the Categorical is not sortable and you would have to explicitly convert each time (`stringvar.astype("category", ordered=True)`)
* even if the groupby var is a Categorical, I'm fine if this takes the a defined categories order and yields the groupby elements by that unless sort is explicitly set to True.